includes/functions.php

```diff
@@ -1478,12 +1478,22 @@ function get_attribution_domains() {
  * @return string The upload base URL.
  */
 function get_upload_baseurl() {
-	$upload_dir = \wp_get_upload_dir();
+	/**
+	 * Early filter to allow plugins to set the upload base URL.
+	 *
+	 * @param string|false $maybe_upload_dir The upload base URL or false if not set.
+	 */
+	$maybe_upload_dir = apply_filters( 'pre_activitypub_get_upload_baseurl', false );
+	if ( false !== $maybe_upload_dir ) {
+		return $maybe_upload_dir;
+	}
+
+	$upload_dir = \wp_upload_dir();
```

Comment:
why not `wp_get_upload_dir()`?? to save a function call?

Reply:
Oh, TIL. Just switching to that would have fixed our issue too.